File: src/main/java/org/mitallast/queue/raft/persistent/PersistentService.java
package org.mitallast.queue.raft.persistent;

import javaslang.control.Option;
import org.mitallast.queue.transport.DiscoveryNode;

public interface PersistentService {

    long currentTerm();

    Option<DiscoveryNode> votedFor();

    void updateState(long newTerm, Option<DiscoveryNode> node);

    ReplicatedLog openLog();
}

File: src/main/java/org/mitallast/queue/raft/persistent/PersistentService.kt
package org.mitallast.queue.raft.persistent

import javaslang.control.Option
import org.mitallast.queue.transport.DiscoveryNode

interface PersistentService {

    fun currentTerm(): Long

    fun votedFor(): Option<DiscoveryNode>

    fun updateState(newTerm: Long, node: Option<DiscoveryNode>)

    fun openLog(): ReplicatedLog
}
